.github/workflows/rust.yml

```diff
@@ -24,22 +24,27 @@ jobs:
     runs-on: ubuntu-latest-16core
     container: ghcr.io/famedly/rust-container:nightly
     steps:
-    - uses: actions/checkout@v3
-    
-    - uses: ./.github/actions/rust-prepare
-      with:
-        gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
-        gitlab_user: ${{ secrets.GITLAB_USER }}
-        gitlab_pass: ${{ secrets.GITLAB_PASS }}
-        additional_packages: ${{ env.ADDITIONAL_PACKAGES }}
+      - uses: actions/checkout@v3
+      
+      - uses: ./.github/actions/rust-prepare
+        with:
+          gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
+          gitlab_user: ${{ secrets.GITLAB_USER }}
+          gitlab_pass: ${{ secrets.GITLAB_PASS }}
+          additional_packages: ${{ env.ADDITIONAL_PACKAGES }}
 
-    - name: Check
-      run: |
-        cargo +${NIGHTLY_VERSION} fmt -- --check
-        cargo lints clippy --workspace --all-targets -- -D warnings
-        cargo llvm-cov nextest --workspace --ignore-filename-regex "cargo_home.*"
-        cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
-        cargo +${NIGHTLY_VERSION} udeps
+      - name: Caching
+        uses: Swatinem/rust-cache@b8a6852b4f997182bdea832df3f9e153038b5191
 
-    - name: Build
-      run: cargo build --verbose --release
+      - name: Check
+        shell: bash
+        run: |
+          cargo +${NIGHTLY_VERSION} fmt -- --check
+          cargo lints clippy --workspace --all-targets -- -D warnings
+          cargo llvm-cov nextest --workspace --ignore-filename-regex "cargo_home.*"
+          cargo +${NIGHTLY_VERSION} test --doc --workspace
+          cargo +${NIGHTLY_VERSION} udeps
```

Comment:
Did we want to split checking, linting and testing into different jobs or steps here?

Reply:
That's the question, do we want this or not? :D It has advantages and disadvantages